invenio_sipstore/archivers/bagit_archiver.py

```diff
@@ -54,25 +63,137 @@ def get_all_files(self):
+        return json.loads(bagit_meta.content)
+
+    def create_bagit_metadata(self, content):
+        """Create the BagIt metadata object."""
+        with db.session.begin_nested():
+            obj = SIPMetadata(sip_id=self.sip.id,
+                              type_id=self.bagit_metadata_type.id,
+                              content=content)
+            db.session.add(obj)
+
+    def create(self, create_bagit_metadata=False, patch_of=None,
+               include_missing_files=False, filesdir="data/files",
+               metadatadir="data/metadata"):
         """Archive the SIP generating a BagIt file.
 
+        When specifying 'patch_of' parameter the 'include_missing_files'
+        flag determines whether files that are missing in the archived SIP
+        (w.r.t. the SIP specified in 'patch_of') should be treated as
+        explicitly deleted (include_missing_files=False) or of all
+        should be included in the manifest.
+
+        Example:
+            include_missing_files = False
+              SIP_1:
+                SIPFiles: a.txt, b.txt
+                BagIt Manifest: a.txt, b.txt
+              SIP_2 (Bagged with patch_of=SIP_1):
+                SIPFIles: b.txt, c.txt
+                BagIt Manifest: b.txt, c.txt
+                fetch.txt: b.txt
+
+            include_missing_files = True
+              SIP_1:
+                SIPFiles: a.txt, b.txt
+                BagIt Manifest: a.txt, b.txt
+              SIP_2 (Bagged with patch_of=SIP_1):
+                SIPFiles: b.txt, c.txt
+                BagIt Manifest: a.txt, b.txt, c.txt
+                fetch.txt: a.txt, b.txt
+
+        :param bool create_bagit_metadata: At the end of archiving,
+            create a SIPMetadata object for this SIP in the DB, which
+            will contain the metadata on the BagIt contents.
+        :param patch_of: Write a lightweight BagIt, which will
+            write only the new files between the previous SIP, and refer to the
+            repeated ones in "fetch.txt" file.
+            The provided argument is a SIP, which will be taken as a base
+            for determining the "diff" between two sets of archives.
+            Provided SIP needs to have a special 'bagit' SIPMetadata object
+            associated with it, i.e. it had to been previously archived with
+            the flag 'create_bagit_metadata'.
+        :type patch_of: invenio_sipstore.models.SIP or None
+        :type bool include_missing_files: If True and when 'patch_of' is used,
+            include the files that are missing in the SIP w.r.t. to
+            the 'patch_of' SIP in the manifest.
+            The opposite (include_missing_files=False) is equivalent to
+            treating those as explicitly deleted.
         :returns: a dictionnary with the filenames as keys, and size and
             checksum as value
         :rtype: dict
         """
+        sipfiles = self.sip.files
+        if patch_of:
+            fetch_info = {}
```

Comment:
Throughout the next operations it's a bit unclear what's the general format/contents of the `fetch_info` dict. Maybe it's worth giving a brief "schema" definition or example here (or referencing somehow the output format from `self._copy_files` I guess...)